app/models/git_repository.rb

```diff
@@ -1,31 +1,39 @@
 class GitRepository
   attr_reader :repository_url, :repository_directory
+  attr_accessor :executor
 
   # The directory in which repositories should be cached.
   cattr_accessor(:cached_repos_dir, instance_writer: false) do
     Rails.application.config.samson.cached_repos_dir
   end
 
-  def initialize(repository_url:, repository_dir:)
+  def initialize(repository_url: nil, repository_dir: nil, executor: nil)
+    raise ArgumentError.new("GitRepository.new: repository_url is required") if repository_url.blank?
```

Comment:
Ah, I didn't realize the `repository_url:,` syntax meant that it was required.  Will revert.